src/app-layer-ssl.c

```diff
@@ -2937,20 +2943,26 @@ void RegisterSSLParsers(void)
         SCLogDebug("ssl_config.encrypt_mode %u", ssl_config.encrypt_mode);
 
         /* Check if we should generate JA3 fingerprints */
+        int enable_ja3 = SSL_CONFIG_DEFAULT_JA3;
         if (ConfGetBool("app-layer.protocols.tls.ja3-fingerprints",
-                        &ssl_config.enable_ja3) != 1) {
-            ssl_config.enable_ja3 = SSL_CONFIG_DEFAULT_JA3;
+                        &enable_ja3) != 1) {
+            enable_ja3 = SSL_CONFIG_DEFAULT_JA3;
+        } else {
+            if (enable_ja3 == 0) {
+                ssl_config.disable_ja3 = true;
+            }
         }
```

Comment:
I think we need to support an "auto" value for this configuration field.  We had true, false and auto if not defined. For completeness, you should be able to get the undefined behaviour with a specific value as well.

Reply:
Yeah can add that.